interop-testing/src/main/java/io/grpc/testing/integration/StressTestClient.java

```diff
@@ -0,0 +1,574 @@
+
+    for (String[] nameWeight : parseCommaSeparatedTuples(testCasesStr)) {
+      TestCases testCase = TestCases.fromString(nameWeight[0]);
+      int weight = Integer.valueOf(nameWeight[1]);
+      testCaseWeightPairs.add(new TestCaseWeightPair(testCase, weight));
+    }
+
+    return testCaseWeightPairs;
+  }
+
+  private static List<String[]> parseCommaSeparatedTuples(String str) {
+    List<String[]> tuples = new ArrayList<String[]>();
+
+    String[] tuples0 = str.split(",");
+    for (String tupleStr : tuples0) {
+      String[] tuple = tupleStr.split(":");
+      if (tuple.length != 2) {
+        throw new IllegalArgumentException("Illegal tuple format: " + tupleStr);
+      }
+      tuples.add(tuple);
+    }
+
+    return tuples;
+  }
+
+  private static ManagedChannel createChannel(InetSocketAddress address) {
+    return NettyChannelBuilder.forAddress(address)
+        .negotiationType(NegotiationType.PLAINTEXT)
+        .build();
+  }
+
+  private static String serverAddressesToString(List<InetSocketAddress> addresses) {
+    return Joiner.on(",").join(Lists.transform(addresses,
+        new Function<InetSocketAddress, String>() {
+          @Nullable
+          @Override
+          public String apply(@Nullable InetSocketAddress input) {
+            Preconditions.checkNotNull(input);
+            return input.getHostName() + ":" + input.getPort();
+          }
+        }));
+  }
+
+  private static String validTestCasesHelpText() {
+    StringBuilder builder = new StringBuilder();
+    for (TestCases testCase : TestCases.values()) {
+      String strTestcase = testCase.name().toLowerCase();
+      builder.append("\n      ")
+          .append(strTestcase)
+          .append(": ")
+          .append(testCase.description());
+    }
+    return builder.toString();
+  }
+
+  /**
+   * A stress test worker. Every stub has its own stress test worker.
+   */
+  private class Worker implements Runnable {
+
+    // Interval at which the QPS stats of metrics service are updated.
+    private static final long METRICS_COLLECTION_INTERVAL_SECS = 5;
+
+    private final ManagedChannel channel;
+    private final List<TestCaseWeightPair> testCaseWeightPairs;
+    private final int durationSec;
+    private final String gaugeName;
+
+    Worker(ManagedChannel channel, List<TestCaseWeightPair> testCaseWeightPairs,
+        int durationSec, String gaugeName) {
+      this.channel = channel;
+      this.testCaseWeightPairs = testCaseWeightPairs;
+      this.durationSec = durationSec;
+      this.gaugeName = gaugeName;
+    }
+
+    @Override
+    public void run() {
+      // Simplify debugging if the worker crashes / never terminates.
+      Thread.currentThread().setName(gaugeName);
+
+      final Tester tester = new Tester();
+      tester.setUp();
+      final WeightedTestCaseSelector testCaseSelector =
+          new WeightedTestCaseSelector(testCaseWeightPairs);
+      final long endTime = durationSec < 0
```

Comment:
If you want to indicate a possibly absent value, use either a boxed primitive or Optional.